Marlin/src/pins/pins_RAMPS_RE_ARM.h

```diff
@@ -195,12 +195,15 @@
     #define SPINDLE_DIR_PIN           P1_19   // (5)
   #endif
 #endif
+
 //
 // Průša i3 MK2 Multiplexer Support
 //
-#define E_MUX0_PIN         P0_03   // ( 0) Z_CS_PIN
-#define E_MUX1_PIN         P0_02   // ( 1) E0_CS_PIN
-#define E_MUX2_PIN         P0_26   // (63) E1_CS_PIN
+#if ENABLED(MK2_MULTIPLEXER)
+  #define E_MUX0_PIN         P0_03   // ( 0) Z_CS_PIN
+  #define E_MUX1_PIN         P0_02   // ( 1) E0_CS_PIN
+  #define E_MUX2_PIN         P0_26   // (63) E1_CS_PIN
+#endif
```

Comment:
There's no need to wrap pin definitions for disabled features. This just clutters the pins files, whose purpose is to declare the pins that would be used for the feature, if it were enabled. We've been gradually scrubbing the code to remove all instances where the existence of a pin implies that a feature is enabled so that we can apply this paradigm throughout and keep pins files as flat as possible.

Reply:
This was done because P0_03, P0_02 will not be defined if the associated serial port is used, an explicit compile error was preferred over silent conflict (defining them as NC), but that only makes sense if the feature is enabled.

Reply:
It's a conundrum. It means that in `pinsDebugging` we have to add `#if ENABLED()` clauses for pins that apply to features, so that the report shows only the meaningful pins. Silent conflicts are okay, and in fact encouraged. Pins can overlap all day as long as only one of their associated features is enabled.

We now look at the pins files as specifying the _recommended_ pins. In some cases a pins file will look for a feature like `SDSUPPORT` and swap some pins. But it's still just a strong suggestion. The ones in `#ifdef` clauses are just the looser ones that we expect to be changed as part of configuration.

Other approaches may be taken, such as, for example, defining `DFLT_BEEPER_PIN` and then only later if `BEEPER_PIN` has never been assigned, set it to `DFLT_BEEPER_PIN`. This has a certain appeal, but would be a huge overhaul to the pins files.

To avoid needing to change pins files, all the code could instead be altered to use `MF_BEEPER_PIN`, which would be assigned only after all pins and configs files were processed. If `MF_BEEPER_PIN` was not already set, it would be set to `BEEPER_PIN`. To override the beeper pin, you would define `MF_BEEPER_PIN` in any configuration file.

Adding another tier to the pins schema may seem crazy, but it would afford another opportunity to override pins in a clean manner. The meaning of a pin macro doesn't crystallize until the macro is expanded, so you want to avoid referencing or using the pin name until the last possible point. An extra tier would ensure that all pins are gathered and ready to be used before any code or preprocessor logic tries to evaluate them.

It would be nice if we had all the pins in a database and we could manipulate, shuffle, and compare them at a higher level, do error checking, produce new pins files as output, and so on….